evm.md

```diff
@@ -2129,6 +2186,44 @@ There are several helpers for calculating gas (most of them also specified in th
 
     rule [Cmem]: Cmem(SCHED, N) => (N *Int Gmemory < SCHED >) +Int ((N *Int N) /Int Gquadcoeff < SCHED >)
 
+    rule [Caddraccess]:
+         Caddraccess(SCHED, ISWARM)
+      => #if ISWARM #then Gwarmstorageread < SCHED > #else Gcoldaccountaccess < SCHED > #fi
+
+    rule [Cstorageaccess]:
+         Cstorageaccess(SCHED, ISWARM)
+      => #if ISWARM #then Gwarmstorageread < SCHED > #else Gcoldsload < SCHED > #fi
+
+    rule [Csload.new]: Csload(SCHED, ISWARM)  => Cstorageaccess(SCHED, ISWARM) requires Ghasaccesslist << SCHED >>
+    rule [Csload.old]: Csload(SCHED, _ISWARM) => Gsload < SCHED >              requires notBool Ghasaccesslist << SCHED >>
+
+    rule [Cextcodesize.new]: Cextcodesize(SCHED, ISWARM)  => Caddraccess(SCHED, ISWARM) requires Ghasaccesslist << SCHED >>
+    rule [Cextcodesize.old]: Cextcodesize(SCHED, _ISWARM) => Gextcodesize < SCHED >     requires notBool Ghasaccesslist << SCHED >>
+
+    rule [Cextcodehash.new]: Cextcodehash(SCHED, ISWARM)  => Caddraccess(SCHED, ISWARM) requires Ghasaccesslist << SCHED >>
+    rule [Cextcodehash.old]: Cextcodehash(SCHED, _ISWARM) => Gbalance < SCHED >         requires notBool Ghasaccesslist << SCHED >>
+
+    rule [Cbalance.new]: Cbalance(SCHED, ISWARM)  => Caddraccess(SCHED, ISWARM) requires Ghasaccesslist << SCHED >>
+    rule [Cbalance.old]: Cbalance(SCHED, _ISWARM) => Gbalance < SCHED >         requires notBool Ghasaccesslist << SCHED >>
+
+    rule [Cextcodecopy.new]: 
+        Cextcodecopy(SCHED, WIDTH, ISWARM)
+      => Caddraccess(SCHED, ISWARM) +Int (Gcopy < SCHED > *Int (WIDTH up/Int 32))
+    requires Ghasaccesslist << SCHED >>
+    rule [Cextcodecopy.old]:
+         Cextcodecopy(SCHED, WIDTH, _ISWARM)
+      => Gextcodecopy < SCHED > +Int (Gcopy < SCHED > *Int (WIDTH up/Int 32))
+      requires notBool Ghasaccesslist << SCHED >>
+
+    rule [Cmodexp.old]:
+         Cmodexp(SCHED, DATA, BASELEN, EXPLEN, MODLEN)
+      => #multComplexity(maxInt(BASELEN, MODLEN)) *Int maxInt(#adjustedExpLength(BASELEN, EXPLEN, DATA), 1) /Int Gquaddivisor < SCHED >
+      requires notBool Ghasaccesslist << SCHED >>
+    rule [Cmodexp.new]:
+         Cmodexp(SCHED, DATA, BASELEN, EXPLEN, MODLEN)
+      => #calculateGasCost(BASELEN, MODLEN, EXPLEN, #asWord(DATA [96 +Int BASELEN .. EXPLEN]))
```

Comment:
There are some modexp tests which crash at `DATA [96 +Int BASELEN .. EXPLEN]`
From the transaction data looks like the `EXPLEN` is a very large number.
```
"0x00000000000000000000000000000000000000000000000000000000000000ff //BASELEN
   2a1e530000000000000000000000000000000000000000000000000000000000 //EXPLEN
   0000000000000000000000000000000000000000000000000000000000000000" //MODLEN
```

stacktrace
```
#0  0x00007ffff6c78c70 in ?? () from /lib/x86_64-linux-gnu/libgcc_s.so.1
#1  0x00007ffff6c7a0bb in _Unwind_RaiseException () from /lib/x86_64-linux-gnu/libgcc_s.so.1
#2  0x00007ffff6e7d78c in __cxa_throw () from /lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x00000000003532d0 in hook_BYTES_padRight (b=<optimized out>, len=<optimized out>, v=<optimized out>)
    at ./llvm-backend/src/main/native/llvm-backend/runtime/strings/bytes.cpp:178
#4  0x0000000000315f39 in apply_rule_6687 (VarWIDTH=0x1)
    at /home/anvacaru/rv/evm-semantics/.build/usr/lib/kevm/include/kframework/evm-types.md:568
#5  0x000000000031078f in EVM.Cmodexp.new.rhs (VarBASELEN=0x7ffff4880068, VarDATA=0x7ffff7ffd9f0 <_rtld_global+2480>, 
    VarEXPLEN=0x7ffff4880198, VarMODLEN=0x7ffff48802a0)
    at /home/anvacaru/rv/evm-semantics/.build/usr/lib/kevm/include/kframework/evm.md:2223
#6  0x00000000002d647d in apply_rule_6041 (Var'Unds'23=<optimized out>, Var'Unds'24=<optimized out>, Var'Unds'25=<optimized out>, 
    Var'Unds'33=<optimized out>, Var'Unds'DotVar0=<optimized out>, Var'Unds'DotVar2=<optimized out>, VarDATA=<optimized out>, 
    VarSCHED=<optimized out>) at /home/anvacaru/rv/evm-semantics/.build/usr/lib/kevm/include/kframework/evm.md:2071
#7  0x0000000000347f62 in take_steps ()
#8  0x000001c500000001 in ?? ()
#9  0x00007ffff487f5d8 in ?? ()
#10 0x000000000034aff9 in main ()
```